biz.aQute.launchpad/src/aQute/launchpad/BundleSpecBuilder.java

```diff
@@ -939,8 +952,16 @@ default BundleSpecBuilder addConfiguration(String config) {
 		return this;
 	}
 
+	/**
+	 * Adds a configuration compatible as described by the Configurator spec. It
+	 * will automatically add the necessary Requirement Header.
+	 *
+	 * @param url the URL, where the Configuration can be found
+	 * @return the current version of the {@link BundleSpecBuilder}
+	 */
 	default BundleSpecBuilder addConfiguration(URL url) {
-		return addResource(CONFIGURATION_JSON, url);
+		return requireCapability("osgi.extender").filter("(osgi.extender=osgi.configurator)")
```

Comment:
if you want to use the constants 
"osgi.configurator"= org.osgi.service.configurator.ConfiguratorConstants.CONFIGURATOR_EXTENDER_NAME
"osgi.extender" = org.osgi.namespace.extender.ExtenderNamespace.EXTENDER_NAMESPACE

Reply:
Good point! I will use the Constants instead.

Reply:
Make sure NOT to use osgi.cmpn or osgi.enterprise to get access to the configurator constant.

Reply:
@bjhargrave Silly, question: what else can I use? Where do you usually get the constants from?

Reply:
The individual OSGi spec jars: https://search.maven.org/artifact/org.osgi/org.osgi.service.configurator/1.0.0/jar